packages/connect-core/src/entities/ForwardingPath.ts

```diff
@@ -0,0 +1,77 @@
+import { ethers } from 'ethers'
+
+import App from './App'
+import Transaction from './Transaction'
+import {
+  AppOrAddress,
+  ForwardingPathData,
+  ForwardingPathDescriptionData,
+  PostProcessDescription,
+} from '../types'
+import { describeForwardingPath } from '../utils/description'
+
+export default class ForwardingPath {
+  #provider: ethers.providers.Provider
+  readonly apps: App[]
+  readonly destination: App
+  readonly transactions: Transaction[]
+
+  constructor(data: ForwardingPathData, provider: ethers.providers.Provider) {
+    this.#provider = provider
+    this.apps = data.apps
+    this.destination = data.destination
+    this.transactions = data.transactions
+  }
+
+  // Lets consumers pass a callback to sign any number of transactions.
+  // This is similar to calling transactions() and using a loop, but shorter.
+  // It returns the value returned by the library, usually a transaction receipt.
+  async sign<Receipt>(
+    callback: (tx: Transaction) => Promise<Receipt>
+  ): Promise<Receipt[]> {
+    return Promise.all(this.transactions.map(async (tx) => await callback(tx)))
+  }
+
+  // Return a description of the forwarding path, to be rendered.
+  async describe(): Promise<ForwardingPathDescription> {
+    // TODO: Make sure we are safe to only provide the apps on the path here
```

Comment:
Here I decide to use the apps that form the path but I wonder if we might need to know the whole list of installed apps. If that case we need to construct the `ForwardingPath` with a reference to the organization so we can fetch information like the list of apps.